src/components/SideBar/SideBar.js

```diff
@@ -0,0 +1,75 @@
+import { SideBarHeader, SideBarList } from "./index.js";
+import {
+    deleteDocument,
+    getRootDocument,
+    makeNewDocument,
+} from "../../service/api.js";
+import { push } from "../../router.js";
+import { removeItem } from "../../utils/storage.js";
+import checkNewComponent from "../../utils/checkNewComponent.js";
+
+export default function SideBar({ $target }) {
+    const self = this;
+    checkNewComponent(SideBar, self);
+
+    const $sideBar = document.createElement("div");
+    $sideBar.className = "sidebar";
+    new SideBarHeader({ $target: $sideBar });
+
+    const sideBarList = new SideBarList({
+        $target: $sideBar,
+        initialState: [],
+        addSubDocument: async (documentId) => {
+            const newDocument = await makeNewDocument("", {
+                title: null,
+                parent: documentId,
+            });
+            const { id } = newDocument;
+            push(`/documents/${id}`);
+
+            self.render();
+        },
+        deleteCurrDocument: async (documentId) => {
+            await deleteDocument(`/${documentId}`);
+            removeItem(`temp-post-${documentId}`);
+
+            push("/");
+            history.replaceState(null, null, "/");
+
+            self.render();
+        },
+        onClick: async (documentId) => {
+            push(`/documents/${documentId}`);
+        },
+    });
+
+    const $addButton = document.createElement("button");
+    $sideBar.appendChild($addButton);
+    $addButton.className = "addButton";
+
+    $addButton.innerText = `새로운 루트 페이지 추가`;
+    $addButton.addEventListener("click", async (event) => {
+        event.preventDefault();
+
+        const newDocument = await makeNewDocument("", {
+            title: null,
+            parent: null,
+        });
+
+        const { id } = newDocument;
+        push(`/documents/${id}`);
+
+        self.render();
+    });
+
+    const getAllDocument = async () => {
+        const allDocument = await getRootDocument();
+        sideBarList.setState(allDocument);
+    };
```

Comment:
함수의 이름은 "도큐먼트를 가져온다" 뿐이므로

```suggestion
    const getAllDocument = () => {
        return getRootDocument(); // 근데 이러면 그냥 getRootDocument를 쓰는 것과 동치이긴 합니다
    };
```
처럼 상태를 set하는 관심사를 제외해주거나, 이름을 `updateDocumentList`와 같이 상태를 set하는 관심사를 포함하는 이름을 지어주면 좋을 것 같아요 ㅎㅎ

Reply:
좋은 의견 감사합니다!!! 🙌
내부동작과 연관 지어 이름을 더 명확하게 짓도록 노력해보겠습니다 ㅎㅎ 👍👍